isaric/parsers/ccp-senegal.toml

```diff
@@ -0,0 +1,739 @@
+  field = "daily_rrt_cmtrt"
+  ref = "Y/N/NK"
+
+[visit.treatment_ecmo]
+  field = "daily_ecmo_prtrt"
+  ref = "Y/N/NK"
+
+[visit.treatment_noninvasive_ventilation]
+  field = "daily_noninvasive_prtrt"
+  ref = "Y/N/NK"
+
+[visit.treatment_invasive_ventilation]
+  field = "daily_invasive_prtrt"
+  ref = "Y/N/NK"
+
+[visit.treatment_high_flow_nasal_cannula]
+  field = "daily_nasaloxy_cmtrt"
+  ref = "Y/N/NK"
+
+[visit.treatment_inotropes_vasopressors]
+  field = "daily_inotrope_cmyn"
+  ref = "Y/N/NK"
+
+[visit.treatment_neuromuscular_blocking_agents]
+  field = "daily_neuro_cmtrt"
+  ref = "Y/N/NK"
+
+[visit.treatment_inhaled_nitric_oxide]
+  field = "daily_nitritc_cmtrt"
+  ref = "Y/N/NK"
+
+[visit.treatment_antimalarial]
+  combinedType = "any"
+
+  [[visit.treatment_antimalarial.fields]]
+    fieldPattern = "tmt_pdthosp_nom.*"
+
+    [visit.treatment_antimalarial.fields.values]
+      HYDROXYCHLOROQUINE = true
+
+[visit.treatment_antibiotics]
+  combinedType = "any"
+
+  [[visit.treatment_antibiotics.fields]]
+    fieldPattern = "tmt_pdthosp_nom.*"
+
+    [visit.treatment_antibiotics.fields.values]
+      "ACIDE CLAVULANIQUE" = true
+      AMOXICILLINE = true
+      APEXIME = true
+      AZICURE = true
+      AZITHROMYCINE = true
+      C-TRI = true
+      "C.TRI" = true
+      CEFTRIAXONE = true
+      CIPROFLOXACINE = true
+      IMIPENEME = true
+      NOVOCLIN = true
+      TINIDAZOLE = true
+      "TOBREX COLLYRE" = true
+
+[visit.treatment_antifungal_agent]
+  combinedType = "any"
+
+  [[visit.treatment_antifungal_agent.fields]]
+    fieldPattern = "tmt_pdthosp_nom.*"
+
+    [visit.treatment_antifungal_agent.fields.values]
+      FLUCOZOLE = true
+      HEXETIDINE = true
+      "MYCOSTER 1%" = true
+
+[visit.treatment_nsaid]
+  combinedType = "any"
+
+  [[visit.treatment_nsaid.fields]]
+    fieldPattern = "tmt_pdthosp_nom.*"
+
+    [visit.treatment_nsaid.fields.values]
+      AINS = true
+      ASPEGIC = true
+      ASPIRINE = true
+
+[visit.treatment_other]
+  combinedType = "set"
+  excludeWhen = "none"
+  fields = [{ field = "daily_other_prtrt" }, { field = "other_cmtrt" }]
+  description = "Other treatment"
+
+[visit.treatment_oxygen_therapy]
+  combinedType = "any"
+  fields = [
+    { field = "oxygen_proccur", ref = "Y/N/NK" },
+    { field = "daily_nasaloxy_cmtrt", ref = "Y/N/NK" },
+    { field = "daily_invasive_prtrt", ref = "Y/N/NK" },
+    { field = "daily_noninvasive_prtrt", ref = "Y/N/NK" },
+  ]
+
+  # observation
+
+[[observation]]
+  name = "avpu"
+  date = { ref = "dailyDate" }
+  phase = "study"
+  text = { field = "avpu_vsorres", values = { 1 = "Alert", 2 = "Verbal", 3 = "Pain", 4 = "Unresponsive" } }
+
+[[observation]]
+  name = "glasgow_coma_score"
+  date = { ref = "dailyDate" }
+  phase = "study"
+  value = { field = "daily_gcs_vsorres" }
+
+[[observation]]
+  name = "cough"
+  date = { ref = "dailyDate" }
+  phase = "study"
+  is_present = { field = "daily_cough_ceoccur", ref = "Y/N/NK" }
```

Comment:
If 'cough' is an upper level hierarchy, make combined type with the +sputum/+haemoptysis values that share the same date

Reply:
Done!